src/compiler/parser.ts

```diff
@@ -9711,8 +9712,13 @@ namespace Parser {
                 if (isBracketed) {
                     skipWhitespace();
                 }
-                const name = parseJSDocIdentifierName(Diagnostics.Unexpected_token_A_type_parameter_name_was_expected_without_curly_braces);
 
+                let modifiers: Modifier[] | undefined;
+                if (token() === SyntaxKind.ConstKeyword) {
+                    modifiers = append(modifiers, parseTokenNode<ConstKeyword>());
+                }

```

Comment:
Shouldn't this do the same thing that `parseTypeParameter` already does?
```suggestion
                const modifiers = parseModifiers(/*allowDecorators*/ false, /*permitConstAsModifier*/ true);
```

The advantage of this would likely be that you'd also handle `in` and `out` modifiers.

Reply:
It would also allow parsing "incorrect" modifiers - but TS already allows this in `.ts` files. The diagnostics about incorrect modifier usage are raised later - but the parsing alone succeeds with incorrect ones. Like for example:
```ts
function test<private T>() {} // 'private' modifier cannot appear on a type parameter(1273)
```